### ISSUE TITLE
Image dimensions in output do not match size argument

### ISSUE THREAD

**Issue by joakimsahlgren** [Peltarion/sidekick#19]
**Describe the bug**
The dimensions of the images in the output from `sidekick.create_dataset()` do not match the size parameter. This probem occurs when the mode parameter is set to `center_crop_or_pad`.

**To Reproduce**
1. Create a new folder.
1. Create one or two subfolders and copy some images to these.
2. Set `input_path` and `output_path` below and then run the code. The `input_path`should point to the newly created top-level folder.
```python
import functools
import sidekick
import os
from glob import glob
from PIL import Image
import pandas as pd

input_path = '/tmp/input'
output_path = '/tmp/output/data.zip'

# Include images of type jpg and png
images_full_path = glob(os.path.join(input_path, '*', '*.jpg')) \
+ glob(os.path.join(input_path, '*', '*.png'))
images = []
classes = []
im_filter = 0
for i in images_full_path:
    i_rel_path = os.path.join(*i.split('/')[-2:]) # path including 'class/file'
    class_name = i_rel_path.split('/')[0]
    # Open image, resize and save in new path
    im = Image.open(i)    
    if im.mode not in ['RGB']:
        im_filter += 1
        print('Filtering: ' + i_rel_path)
        continue         
    im.close()
    img_path = os.path.join(input_path, i_rel_path)
    images.append(img_path)
    classes.append(class_name)
df = pd.DataFrame({'image': images, 'class': classes})
# Create the dataset
image_processor = functools.partial(sidekick.process_image, mode='center_crop_or_pad', size=(299, 299), format='jpeg')
sidekick.create_dataset(
    output_path,
    df,
    path_columns=['image'],
    preprocess={
        'image': image_processor
    }    
)
```
3. Unzip the output file.
4. Check the dimension of the output file. 

**Expected behavior**
All the output file should have the dimensions 299x299.

**Desktop (please complete the following information):**
 - OS: macOS Mojave 10.14.4
 - Packages in your environment:
Package              Version 
-------------------- --------
absl-py              0.6.1   
astor                0.7.1   
certifi              2019.3.9
chainer              5.1.0   
chardet              3.0.4   
elasticsearch        6.3.1   
filelock             3.0.10  
gast                 0.2.0   
grpcio               1.17.1  
h5py                 2.9.0   
idna                 2.8     
imagecrawler         0.2     
ImageHash            4.0     
Keras-Applications   1.0.6   
Keras-Preprocessing  1.0.5   
Markdown             3.0.1   
mock                 2.0.0   
numpy                1.15.4  
opencv-python        3.4.4.19
pandas               0.24.1  
pbr                  5.1.3   
Pillow               5.4.1   
pip                  18.1    
protobuf             3.6.1   
python-dateutil      2.8.0   
pytz                 2018.9  
PyWavelets           1.0.2   
requests             2.21.0  
scipy                1.2.1   
selenium             3.141.0 
setuptools           40.5.0  
sidekick             0.1.0   
six                  1.12.0  
tensorboard          1.13.1  
tensorflow           1.13.1  
tensorflow-estimator 1.13.0  
termcolor            1.1.0   
tqdm                 4.31.1  
urllib3              1.24.1  
virtualenv           16.1.0  
Werkzeug             0.14.1  
wheel                0.32.2  
zipfile36            0.1.3   

**Additional context**

**Comment by agrinh** [comment]
Thanks for creating this issue! Can you provide some example images where this happens?

**Comment by joakimsahlgren** [comment]
Examples images:
[samples.zip](https://github.com/Peltarion/sidekick/files/3057748/samples.zip)

Example 1 - 500x700 pixels:
![broken_back_light_116](https://user-images.githubusercontent.com/45963830/55783264-429e7100-5aae-11e9-93ce-644e3b58f194.jpg)

Example 1 (after preprocessing) - 298x298:
![692 2](https://user-images.githubusercontent.com/45963830/55783368-8abd9380-5aae-11e9-9673-e99c37f1bc79.jpeg)

**Comment by agrinh** [comment]
@joakimsahlgren this should have been fixed in #21, can you confirm that it solved your issue?

**Comment by joakimsahlgren** [comment]
> @joakimsahlgren this should have been fixed in #21, can you confirm that it solved your issue?

Confirmed.

**Comment by agrinh** [comment]
Thanks @joakimsahlgren and @alvaro-garcia-carrasco, closing!
